packages/core/src/utils/JsonTransformer.ts

```diff
@@ -1,15 +1,41 @@
+import type { Validate } from 'class-validator'
+
 import { instanceToPlain, plainToInstance, instanceToInstance } from 'class-transformer'
 
+import { ClassValidationError } from '../error/ClassValidationError'
+
+import { MessageValidator } from './MessageValidator'
+
+interface Validate {
+  validate?: boolean
+}
+
 export class JsonTransformer {
   public static toJSON<T>(classInstance: T) {
     return instanceToPlain(classInstance, {
       exposeDefaultValues: true,
     })
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  public static fromJSON<T>(json: any, Class: { new (...args: any[]): T }): T {
-    return plainToInstance(Class, json, { exposeDefaultValues: true })
+  public static fromJSON<T>(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    json: any,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    Class: { new (...args: any[]): T },
+    { validate = true }: Validate = {}
+  ): T {
+    const instance = plainToInstance(Class, json, { exposeDefaultValues: true })
+
+    // Skip validation
+    if (!validate === true) return instance
+
+    if (instance === undefined || instance === null) {
+      throw new ClassValidationError('Cannot validate instance of ', { classType: typeof instance })
+    }
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    MessageValidator.validateSync(instance, Class as any)
```

Comment:
```suggestion
    MessageValidator.validateSync(instance, Class as T)
```

I am not too sure if this would work

Reply:
Happy to change that otherwise but casting to T gives:

```
Argument of type 'T' is not assignable to parameter of type 'new (...args: any[]) => object'
```
ts(2345)